### PR TITLE
Discord OAuth support

### PR DESCRIPTION
**Pull request by BytewaveMLP**
- Add `passport-discord`
  - Update `npm` dependencies so we actually have the Discord FA icon

TODO:

  - Pretty up code a bit; maybe get rid of that nasty ternary
  - See what can be done about Discord profile links

### CODE REVIEW

File: src/ts/graphics/webgl/webglUtils.ts [hunk 1]
```diff
@@ -18,9 +18,9 @@ export function getWebGLContext(canvas: HTMLCanvasElement): WebGLRenderingContex
 		antialias: false,
 	};
 
-	const gl = canvas.getContext('webgl2', options)
+	const gl = (canvas.getContext('webgl2', options)
 		|| canvas.getContext('webgl', options)
-		|| canvas.getContext('experimental-webgl', options);
+		|| canvas.getContext('experimental-webgl', options)) as WebGLRenderingContext;
```
Comment (BytewaveMLP) [inline src/ts/graphics/webgl/webglUtils.ts]: There's a fix for this in `src/typings/my/fix.d.ts` which appears to have been broken as a result of the `npm upgrade`.

cc @CyberPon3, typescript god pls help

**Comment by BytewaveMLP** [comment]
I think this is close to a state where it can be merged. There's a couple of extra things I'd like to look into, including some long-term changes to the OAuth infrastructure as a whole, but for now this is capable enough.

File: README.md [hunk 2]
```diff
@@ -107,6 +107,21 @@ Get OAuth keys for authentication platform of your choice (github, google, twitt
 },
 ```
 
+### Discord
+
+- Go to https://discordapp.com/developers/applications/ and create a new app
+- Navigate to the OAuth2 tab
+- Add `http://<your domain>/auth/discord/callback` (or `http://localhost:8090/auth/vkontakte/callback` for your localhost server) as a redirect URI
```
Comment (blackhole12) [inline README.md]: pretty sure the URL here shouldn't be for `vkontakte`

File: src/ts/graphics/webgl/webglUtils.ts (hunk 1 above)
Comment (blackhole12) [reply src/ts/graphics/webgl/webglUtils.ts]: Has this been resolved?

File: README.md (hunk 2 above)
Comment (BytewaveMLP) [reply README.md]: Whoops, that's what I get for c/ping.

File: src/ts/graphics/webgl/webglUtils.ts (hunk 1 above)
Comment (BytewaveMLP) [reply src/ts/graphics/webgl/webglUtils.ts]: Looks like TS<3.6 will fix this.

Comment (BytewaveMLP) [reply src/ts/graphics/webgl/webglUtils.ts]: Confirmed. fa0271c

**Comment by BytewaveMLP** [comment]
Angular/TS weirdness resolved with a downgrade to sub-3.6 TypeScript.

**Pull request merged**